data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CSVProcessor.java

```diff
@@ -45,4 +83,65 @@ public boolean isReadyForShutdown() {
     public void shutdown() {
 
     }
+
+    private CsvMapper createCsvMapper() {
+        final CsvMapper mapper = new CsvMapper();
+        mapper.enable(CsvParser.Feature.WRAP_AS_ARRAY); // allows mapper to read with empty schema
+        return mapper;
+    }
+
+    private CsvSchema createCsvSchema() {
+        final char delimiterAsChar = config.getDelimiter().charAt(0); // safe due to config input validations
+        final char quoteCharAsChar = config.getQuoteCharacter().charAt(0); // safe due to config input validations
+        final CsvSchema schema = CsvSchema.emptySchema().withColumnSeparator(delimiterAsChar).withQuoteChar(quoteCharAsChar);
+        return schema;
+    }
+
+    private List<String> parseHeader(Event event, final boolean thisEventHasHeaderSource, final CsvMapper mapper,
```

Comment:
nit: missed a couple `final`s for method arguments. `event` here and in `putDataInEvent` and `colNumber` for `generateColumn`. Exceptions within a `catch` clause can be `final` too - `catch (final IOException e) {`